nomos-services/storage/src/lib.rs

```diff
@@ -303,41 +301,40 @@ where
     RuntimeServiceId: AsServiceId<Self> + Display + Send,
 {
     fn init(
-        service_state: OpaqueServiceStateHandle<Self, RuntimeServiceId>,
-        _init_state: Self::State,
+        service_resources_handle: OpaqueServiceResourcesHandle<Self, RuntimeServiceId>,
+        _initial_state: Self::State,
     ) -> Result<Self, DynError> {
         Ok(Self {
-            backend: Backend::new(service_state.settings_reader.get_updated_settings())?,
-            service_state,
+            backend: Backend::new(
+                service_resources_handle
+                    .settings_handle
+                    .notifier()
+                    .get_updated_settings(),
+            )?,
+            service_resources_handle,
         })
     }
 
-    async fn run(mut self) -> Result<(), overwatch::DynError> {
+    async fn run(mut self) -> Result<(), DynError> {
         let Self {
             mut backend,
-            service_state:
-                OpaqueServiceStateHandle::<Self, RuntimeServiceId> {
-                    mut inbound_relay,
-                    lifecycle_handle,
-                    ..
+            service_resources_handle:
+                OpaqueServiceResourcesHandle::<Self, RuntimeServiceId> {
+                    mut inbound_relay, ..
                 },
         } = self;
-        let mut lifecycle_stream = lifecycle_handle.message_stream();
         let backend = &mut backend;
-        loop {
-            tokio::select! {
-                Some(msg) = inbound_relay.recv() => {
-                    Self::handle_storage_message(msg, backend).await;
-                }
-                Some(msg) = lifecycle_stream.next() => {
-                    if lifecycle::should_stop_service::<Self, RuntimeServiceId>(&msg) {
-                        // TODO: Try to finish pending transactions if any and close connections properly
-                        break;
-                    }
-                }
-            }
+
+        while let Some(msg) = inbound_relay.recv().await {
+            Self::handle_storage_message(msg, backend).await;
         }
+
         Ok(())
+        // TODO: Implement `Drop` to finish pending transactions and close
+        //  connections gracefully.
+        // Note: Using `Drop` here would only cover the graceful shutdown case.
+        // If the `Stop` is triggered by a user-sent signal, a different
+        // yet-not-implemented-mechanism  is required.
```

Comment:
Is it? I mean, this should be dropped anyway, the problem is if we move self...

Reply:
Oh, what I wrote here doesn't make a lot of sense, indeed. I'll double check, just in case.

Reply:
Oh, yeah, this doesn't make a lot of sense, indeed. I think I was assuming some behaviour in a previous iteration of the Lifecycle. Updating.

When would `self` be moved?